migrations/sqlite_versions/2020-06-08_4fb1c8780273_add_unique_constraint_for_platform.py

```diff
@@ -0,0 +1,54 @@
+"""Add unique constraint for Platform
+
+Revision ID: 4fb1c8780273
+Revises: e752bda39400
+Create Date: 2020-05-29 16:37:13.344722
+
+"""
+import geoalchemy2
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "4fb1c8780273"
+down_revision = "b03fb305981b"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("Datafiles", schema=None) as batch_op:
+        batch_op.create_unique_constraint("uq_Datafile_size_hash", ["size", "hash"])
+
+    with op.batch_alter_table("GeometrySubTypes", schema=None) as batch_op:
+        batch_op.create_unique_constraint("uq_GeometrySubType_name_parent", ["name", "parent"])
+        batch_op.drop_constraint("uq_GeometrySubTypes_name", type_="unique")
```

Comment:
Yes I think so. In a previous PR we added unique constraints to the name field for each of the reference tables, but made a mistake in that GeometrySubTypes requires a unique constraint on `name` and `parent`, rather than just `name`. This removes the old constraint just for `name` and adds a new one for both `name` and `parent`. Does that sound right to you?

Reply:
Okay, it sounds right but it means that we need it for PostgreSQL as well. Because there is no `drop_constraint()` for that. I think the name field of `GeometrySubTypes` still has `unique=True` parameter.

Reply:
I realised one more thing, @robintw . It seems like we need one more migration for the `cascade` update. When I run the autogenerate command, it drops current foreign keys and creates new ones with the same name but with an extra parameter: `onupdate="cascade"`.

Reply:
I've just tried creating the migration to add the cascades, and have run into a problem: we get lots of lines like `batch_op.drop_constraint(None, type_="foreignkey")` in the SQLite migration (I haven't tried the Postgres migration yet), where it is trying to delete foreign keys that previously existed. Unfortunately these didn't have a name, as they were created before we had the naming convention.

We then get an error when running the script - presumably caused by the `None` in the name, although the actual error isn't very helpful:
```
File "/Users/robin/anaconda3/envs/pepys/lib/python3.7/site-packages/sqlalchemy/sql/naming.py", line 44, in _column_X
    fk = self.const.elements[idx]
IndexError: list index out of range
```

Reply:
I think we should consider merging/modifying old migration scripts, @robintw . Because we already have huge migrations which are really sensitive. And it's getting more difficult to implement new migrations when we keep old migrations as they were created. What do you think?

Reply:
I found a way to workaround this issue, @robintw . It's possible to give names to unnamed foreign key constraints:
https://alembic.sqlalchemy.org/en/latest/batch.html#dropping-unnamed-or-named-foreign-key-constraints

> In order to overcome this, the Operations.batch_alter_table() method supports a **naming_convention** argument, so that all reflected constraints, including foreign keys that are unnamed, or were named but SQLAlchemy isn’t loading this name, may be given a name

Reply:
Note: I deleted my old comment for misunderstanding.
Created and modified migration script for SQLite is as follows: https://github.com/debrief/pepys-import/pull/405/commits/1b44e2687db1bb2d013079f502568b8902ab9982

Reply:
Great, thanks!